### PR TITLE
Relaxed validation rule for version.

### PR DESCRIPTION
**Pull request by vaskalan**
We now accept both 3 and 4 numbers in api version value.
14.0.0 is allowed and equivalent to 14.0.0.0.

Background:

Ørnulf tried this request to rose-backend:
`http://ol-raird-app-qa1.ssb.no/rose-backend/data-store/metadata/all?dataStore=no.ssb.fdb.m2&version=14.0.0`
from which we can see that version is in format x.y.z, not x.y.z.w

### CODE REVIEW

File: src/main/java/no/microdata/datastore/transformations/VersionUtils.java [hunk 1]
```diff
@@ -20,7 +24,7 @@ public class VersionUtils {
 
 
     public static String toFileVersion(String version) {
-        if (!is4PartSemanticVersion(version)) {
+        if ( !isThreeNumbersVersion(version) && !isFourNumbersVersion(version)) {
```
Comment (DanielElisenberg) [inline src/main/java/no/microdata/datastore/transformations/VersionUtils.java:27]: Can we call this "is4PartSemanticVersion" and "is3PartSemanticVersion"?

File: src/test/java/no/microdata/datastore/transformations/VersionUtilsTest.java [hunk 2]
```diff
@@ -9,19 +9,33 @@
...
     @Test
     void shouldConvertToFileVersionDraft() {
         assertEquals("draft", VersionUtils.toFileVersion("0.0.0.133"));
     }
 
     @Test
-    void shouldRequire4LabelsVersion() {
-        BadRequestException exceptionThatWasThrown = Assertions.assertThrows(BadRequestException.class, () -> {
-            VersionUtils.toFileVersion("1.2.3");
+    void shouldRejectAllButThreeOrFourNumbersVersion() {
```
Comment (DanielElisenberg) [inline src/test/java/no/microdata/datastore/transformations/VersionUtilsTest.java:27]: :100:

**Review by DanielElisenberg — APPROVED** [review]
Looks good 👍🏻 One minor naming comment

File: src/main/java/no/microdata/datastore/transformations/VersionUtils.java (hunk 1 above)
Comment (vaskalan) [reply src/main/java/no/microdata/datastore/transformations/VersionUtils.java:27]: Done!

**Pull request merged**